src/main/java/io/jenkins/plugins/neuvector/NeuVectorWorker.java

```diff
@@ -623,7 +624,113 @@ private void getToken(CloseableHttpClient httpclient) throws AbortException {
             }
             throw new AbortException(causeMessage);
         } catch (IOException e) {
-            String causeMessage = "NeuVector controller connection error.";
+            String causeMessage = "NeuVector controller connection error with token: " + e.getMessage();
+            if (logger != null) {
+                logger.println(causeMessage);
+            }
+            throw new AbortException(causeMessage);
+        }
+    }
+
+    private void logout(CloseableHttpClient httpclient) throws IOException {
+        String uriPathForLogout = "/v1/auth";
+        URI uriForLogout = buildUri(config.getControllerApiUrl(), uriPathForLogout);
+        HttpDelete httpDeleteForLogout = new HttpDelete(uriForLogout);
+        httpDeleteForLogout.addHeader("Content-Type", "application/json");
+        httpDeleteForLogout.addHeader("X-Auth-Token", token);
+        CloseableHttpResponse httpResponseFromLogout = httpclient.execute(httpDeleteForLogout);
+        httpResponseFromLogout.close();
+    }
+
+    public void testApiKey() throws IOException, NoSuchAlgorithmException, KeyStoreException, KeyManagementException, SSLHandshakeException, SSLPeerUnverifiedException {
+        try (CloseableHttpClient httpclient = makeHttpClient()) {
+            getSystemSummary(httpclient);
+        }
+    }
+
+    private void getSystemSummary(CloseableHttpClient httpclient) throws AbortException {
+        String uriPathForValidation = "/v1/scan/repository";
```

Comment:
This function should not be named getSystemSummary() if the URL to call is POST(/v1/scan/repository)

Reply:
I fix it to validateApiKeyWithRepoScan.